main.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import "fmt"
+
+func worker(id int, jobs <-chan int, results chan<- int) {
+	for j := range jobs {
+		go func() {
```

Comment:
Issue: Creating and calling a goroutine for each job inside a worker function is inefficient and can cause a lot of context switching.
Fix: This should be done directly.